docs/adr/0042-evaluate-oauth-implementation.md

```diff
@@ -0,0 +1,73 @@
+## ADR 42: Evaluate Strategy for OAuth Implementation
+
+Status: proposed
+
+Deciders: Vijay Budhram, Wil Clouser, TBD
+
+Date: 2024-03-21
+
+## Context and Problem Statement
+
+FxA has had custom OAuth implementation for many years. However, it is complex and 
+difficult to maintain. Our implementation has been slow to adopt new specifications,  
+security updates and ensure compliant with OAuth 2.0 and OpenID Connect. 
+This ADR considers the pros and cons of various options to update and modernize our OAuth implementation.
+
+## Decision Drivers
+
+* Security: Address vulnerabilities.
+* Maintainability: Simplify updates and maintenance.
+* Compliance: Ensure compatiablity with OAuth 2.0 and OpenID Connect.
+* Transition Risk: Minimize migration risk.
+
+## Considered Options
+
+* Maintain Custom OAuth Implementation
+* Use Node [OpenID Connect Provider](https://github.com/panva/node-oidc-provider)
+* Operate Both in Parallel (Hybrid Approach)
+
+## Decision Outcome
+
+To be determined.
+
+## Option 1: Maintain Custom OAuth Implementation
+
+Pros
+
+* Control: Complete control over OAuth implementation details and customization to specific needs.
+* Integration: No immediate changes or migrations.
+
+Cons
+
+* Maintenance: Ongoing requirement to update and secure the custom implementation.
+* Compliance: Increased risk of falling behind on OAuth standards and security practices without dedicated updates.
```

Comment:
To be fair, OAuth 2.0 is pretty old and hasn't changed much.  It is OIDC that we need to keep up with (presuming we are interested in the features from their newer specs).